semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -1635,85 +1841,91 @@ and map_lambda_literal (env : env) ((v1, v2, v3, v4, v5) : CST.lambda_literal) :
     (* Fake brackets here since the brackets delimit the lambda expression as a
      * whole, not just the statements *)
     (* TODO consider using `in` and the closing bracket as the delimiters *)
-    G.FBStmt (G.Block (G.fake_bracket stmts) |> G.s)
+    G.FBStmt
+      (G.OtherStmtWithStmt
+         ( G.OSWS_Closure,
+           Common.map (fun x -> G.Pa x) captures,
+           G.Block (G.fake_bracket stmts) |> G.s )
+      |> G.s)
   in
   let v5 = (* "}" *) token env v5 in
   let def =
     {
       G.fkind = (G.LambdaKind, v1);
       fparams = params;
-      (* TODO include return type if provided *)
-      frettype = None;
+      frettype = rettype;
       fbody = body;
     }
   in
   G.Lambda def |> G.e
 
 and map_lambda_parameter (env : env) ((v1, v2) : CST.lambda_parameter) :
     G.parameter =
-  let v1 =
+  let attrs =
     match v1 with
-    | Some x -> map_attribute env x
-    | None -> None
+    | Some x -> [ map_attribute env x ]
+    | None -> []
   in
   let v2 =
     match v2 with
-    | `Self_exp tok -> (* "self" *) token env tok |> todo env
-    | `Simple_id x -> G.Param (map_simple_identifier env x |> G.param_of_id)
+    | `Self_exp tok -> (* "self" *) G.OtherParam (("Self", token env tok), [])
+    | `Simple_id x ->
+        G.Param (map_simple_identifier env x |> G.param_of_id ~pattrs:attrs)
     | `Opt_simple_id_simple_id_COLON_opt_param_modifs_poss_impl_unwr_type
         (v1, v2, v3, v4, v5) ->
-        let v1 =
+        (* TODO As with `map_parameter`, we only support internal names right now. *)
+        let _v1 =
           match v1 with
-          | Some x -> map_simple_identifier env x
-          | None -> todo env ()
+          | Some x ->
+              let _todo = map_simple_identifier env x in
```

Comment:
Just pipe this to `todo env` so that it doesn't get overlooked if we encounter this?